hypershift-operator/controllers/scheduler/azure/controller.go

```diff
@@ -0,0 +1,148 @@
+package azure
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	hyperv1 "github.com/openshift/hypershift/api/hypershift/v1beta1"
+	schedulingv1alpha1 "github.com/openshift/hypershift/api/scheduling/v1alpha1"
+	"github.com/openshift/hypershift/hypershift-operator/controllers/hostedcluster"
+	"github.com/openshift/hypershift/support/util"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/workqueue"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	aroScheduler       = "AzureScheduler"
+	schedulerFinalizer = "hypershift.openshift.io/azure-scheduler"
+)
+
+type Scheduler struct {
+	client.Client
+}
+
+func (r *Scheduler) SetupWithManager(mgr ctrl.Manager) error {
+	r.Client = mgr.GetClient()
+	builder := ctrl.NewControllerManagedBy(mgr).
+		For(&hyperv1.HostedCluster{}).
+		WithOptions(controller.Options{
+			RateLimiter:             workqueue.NewTypedItemExponentialFailureRateLimiter[reconcile.Request](1*time.Second, 10*time.Second),
+			MaxConcurrentReconciles: 1,
+		}).
+		Watches(&schedulingv1alpha1.ClusterSizingConfiguration{}, handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, obj client.Object) []reconcile.Request {
+			hostedClusters := &hyperv1.HostedClusterList{}
+			if err := r.List(ctx, hostedClusters); err != nil {
+				return nil
+			}
+			var out []reconcile.Request
+			for _, hc := range hostedClusters.Items {
+				out = append(out, reconcile.Request{NamespacedName: types.NamespacedName{Namespace: hc.Namespace, Name: hc.Name}})
+			}
+			return out
+		})).
+		Named(aroScheduler)
+	return builder.Complete(r)
+}
+
+func (r *Scheduler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	hc := &hyperv1.HostedCluster{}
+	log := ctrl.LoggerFrom(ctx)
+	err := r.Get(ctx, req.NamespacedName, hc)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Info("hostedcluster not found, aborting reconcile")
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, fmt.Errorf("failed to get cluster %q: %w", req.NamespacedName, err)
+	}
+	if !hc.DeletionTimestamp.IsZero() {
+		log.Info("hostedcluster is deleted, cleaning up")
+		if controllerutil.ContainsFinalizer(hc, schedulerFinalizer) {
+			if controllerutil.ContainsFinalizer(hc, hostedcluster.HostedClusterFinalizer) {
+				// Wait until the hosted cluster finalizer is removed
+				return ctrl.Result{}, nil
+			}
+			// Ensure that any placeholder deployment is deleted
```

Comment:
>// Ensure that any placeholder deployment is deleted

I don't see where we are honouring this. It doesn't seem this finalizer is actually doing anything.
Is this a TODO?

Reply:
This comment should be removed it was from previous code , we don't need placeholders for the aro scheduler